dra.gradle

```diff
@@ -0,0 +1,122 @@
+
+apply plugin: 'logstash-dra'
+
+logstash {
```

Comment:
I might be missing context around the logstash build, your goals and dra, so take this with a grain of salt and feel free to override what I'm saying :) Would be very interested to see what @mgreau thinks of it too but he is on PTO right now.

It feels to me that this creates too much duplication and to much of maintenance burden to have this definition duplicated here from the infra repo. It also brings in a lot of code to be able to power this DSL. For all that I'm missing the value it adds or the overall advantages of the approach.  

My understanding is that with DRA we acknowledge that there can be some discrepancies between the artifacts uploaded and the ones release manager expects and this is ok for now and something we can think of addressing across the board lather. With this in mind, the logstash DRA builds could be as simples as a few steps in a build pipeline or a build script : 

- rake artifact:all
- bin/dependencies-report --csv=$DEPENDENCIES_REPORTS_DIR/$DEPENDENCIES_REPORT
- bin/upload_artefacts.sh 

`upload_artefacts.sh` would be a new custom script using the rm container to do the upload. This container already has the DSL bundled so it will fail if it can't find one of the expected artefacts. It would be something like (https://docs.elastic.dev/release/stack/unified-release-process-requirements):
```
# collect the artifacts for use with the unified build
docker run --rm \
  --name release-manager \
  -e VAULT_ADDR \
  -e VAULT_ROLE_ID \
  -e VAULT_SECRET_ID \
  --mount type=bind,readonly=false,src="$PWD/build",target=/artifacts \
  docker.elastic.co/infra/release-manager:latest \
    cli collect \
      --project logstash \
      --branch 8.0 \
      --commit "$(git rev-parse HEAD)" \
      --workflow "snapshot" \
      --version "$(cat version.txt)" \
      --artifact-set main
```
Assuming that your existing CI build already creates the artefacts, you would only need to wire in the upload and maybe the dependency report if you don't already have it and you would be good to go.

Reply:
I understand you, but:
- the existing CI doesn't generate all the artifacts, it run unit and integration tests, so execution of `rake artifact:all` actually. That's invoked only from the `release-maneger`
- the existing DSL is a good declarative way to list all the artifacts that has to be created and medium to verify they exists before running the dockerized `release-manager` to upload. I agree with you that's an overkill and a problem of duplication; that's why I was searching for a reusable Gradle plugin.

Reply:
> This container already has the DSL bundled so it will fail if it can't find one of the expected artefacts

If the containerized release-manager already does the checks on artifacts existence before upload, then I agree with you that the DRA on Logstash CI could be simplified to a bash script that executes the build, report creation and upload with release-manager.

Reply:
> * the existing CI doesn't generate all the artifacts, it run unit and integration tests, so execution of `rake artifact:all` actually. That's invoked only from the `release-maneger`

Note that it is the goal of the DRA project that artifacts are generated by the product CI and not anymore by the release manager. Once this will be achieved the `rake artifact:all` instruction should be removed from the [Logstash release-manager DSL](https://github.com/elastic/infra/blob/3bf5bdb103ffbf202c1c739b7120220062fd074f/cd/release/release-manager/project-configs/master/logstash.gradle#L75-L88) and replaced by commands to put the downloaded artifacts to the good directory, [example for the apm-server project](https://github.com/elastic/infra/blob/3bf5bdb103ffbf202c1c739b7120220062fd074f/cd/release/release-manager/project-configs/master/apm-server.gradle#L32-L48):
```
    // Download the APM Server artifacts built externally by apm-ci from GCS
    externalArtifacts('gcs', 'apm-server')

    buildSnapshotCommands = [
      // move dependency report to expected location
      "mv csvDependencyReports/apm-server/dependencies-${qualifiedVersion}-SNAPSHOT.csv \$DEPENDENCIES_REPORTS_DIR/\$DEPENDENCIES_REPORT",
      // move artifacts to the expected location
      'mkdir -p build/distributions',
      'mv downloads/apm-server/* build/distributions/'
    ]
    buildReleaseCommands = [
      // move dependency report to expected location
      "mv csvDependencyReports/apm-server/dependencies-${qualifiedVersion}.csv \$DEPENDENCIES_REPORTS_DIR/\$DEPENDENCIES_REPORT",
      // move artifacts to the expected location
      'mkdir -p build/distributions',
      'mv downloads/apm-server/* build/distributions/'
    ]
```

I suggest scheduling a Zoom call next week once @mgreau is back to clarify this.

Reply:
`rake artifacts:all` will be executed by Logstash CI (directly from the Jenkins job, or invoking a bash script).

The point here, I think, is: "who is responsible to grant that all artifacts needed to be upload to GCS?".
The paths are 2:
- the Logstash CI when launch `rake artifacs:all` also ensure that all the expected artifacts are generated -or-
- as @alpar-t said, the containerized RM has already the list of artifacts to upload, so if it doesn't find anything expected, it fails. So no need to be checked on Logstash CI side.

Reply:
For now, this is solution 2.